torch/nn/modules/rnn.py

```diff
@@ -18,6 +21,7 @@
 
 
 def apply_permutation(tensor: Tensor, permutation: Tensor, dim: int = 1) -> Tensor:
+    warnings.warn("apply_permutation is deprecated, please use tensor.index_select(dim, permutation) instead")
```

Comment:
nit: use index_select directly so that functions in this file don't trigger this warning.

Reply:
added `_apply_permutation` instead for better code readability